moshi/src/main/java/com/squareup/moshi/Types.java

```diff
@@ -216,6 +217,36 @@ public static boolean equals(@Nullable Type a, @Nullable Type b) {
     }
   }
 
+  /**
+   * @param clazz the target class to read the {@code fieldName} field annotations from.
+   * @param fieldName the target field name on {@code clazz}.
+   * @return a set of {@link JsonQualifier}-annotated {@link Annotation} instances retrieved from
+   *         the targeted field. Can be empty if none are found.
+   */
+  public static Set<? extends Annotation> getFieldJsonQualifierAnnotations(Class<?> clazz,
```

Comment:
Deferring to Jesse on this since the was his suggestion. It might not hurt to have a public utility class somewhere that's in a .internal package, but the current implementation is at least somewhat general use.

Reply:
I’d prefer only public APIs for generated code. They’re effectively public anyway.